src/IECore/FileNameParameter.cpp

```diff
@@ -87,8 +91,11 @@ bool FileNameParameter::valueValid( const Object *value, std::string *reason ) c
 	// extensions check
 	if( extensions().size() )
 	{
+#if BOOST_VERSION >= 108500
+		string ext = boost::filesystem::path(boost::filesystem::path( s->readable())).extension().string();
```

Comment:
I believe this line will work in Boost 1.80 as well, so we can simply replace the old code completely. I think we have one-too-many `boost::filesystem::path()` calls though - we only need one.